lib/Net/Async/Blockchain/ETH.pm

```diff
@@ -234,7 +235,10 @@ sub subscribe {
                     # Skip old checked blocks (e.g. when restarting the node)
                     my $next_block        = await $self->new_blocks_queue->shift;
                     my $next_block_number = Math::BigInt->from_hex($next_block->{params}->{result}->{number});
-                    next if $self->base_block_number and $next_block_number->blt($self->base_block_number);
+                    # to avoid chain re-organization check/emit last 6th block instead of highest block
+                    my $safe_block_number =
+                        ($self->base_block_number) ? Math::BigInt->new($self->base_block_number)->bsub(MAX_SAFE_BLOCK_COUNT) : 0;
+                    next if ($next_block_number->blt($safe_block_number));
```

Comment:
```suggestion
                    # to avoid chain re-organization check/emit last 6th block instead of highest block
                    next if $self->base_block_number and $next_block_number->blt($self->base_block_number - MAX_SAFE_BLOCK_COUNT);
```

Reply:
updated. thanks